test/unit/rh_cloud_http_proxy_test.rb

```diff
@@ -18,6 +19,13 @@ class RhCloudHttpProxyTest < ActiveSupport::TestCase
     assert_equal @global_foreman_proxy_mock, ForemanRhCloud.proxy_setting
   end
 
+  test 'returns empty string in on-prem setup' do
+    ForemanRhCloud.unstub(:with_local_advisor_engine?)
+    ForemanRhCloud.stubs(:with_local_advisor_engine?).returns(true)
+
+    assert_equal '', ForemanRhCloud.proxy_setting
```

Comment:
Isn't this a better assertion?
```suggestion
    assert_empty ForemanRhCloud.proxy_setting
```